pkg/handlers/update_checker_spec.go

```diff
@@ -35,6 +44,91 @@ func (h *Handler) ConfigureAutomaticUpdates(w http.ResponseWriter, r *http.Reque
 		return
 	}
 
+	clientSet, err := k8sutil.GetClientset()
+	if err != nil {
+		logger.Error(err)
+		updateCheckerSpecResponse.Error = "failed to create k8s client"
+		JSON(w, 500, updateCheckerSpecResponse)
+		return
+	}
+
+	selectorLabels := map[string]string{
+		"metadata.name": "kotsadm-confg",
+	}
+	listOpts := metav1.ListOptions{
+		FieldSelector: labels.SelectorFromSet(selectorLabels).String(),
+	}
+
+	configMaps, err := clientSet.CoreV1().ConfigMaps(util.PodNamespace).List(context.TODO(), listOpts)
+	if err != nil && !kuberneteserrors.IsNotFound(err) {
+		logger.Error(err)
+		updateCheckerSpecResponse.Error = "failed to list config maps"
+		JSON(w, 500, updateCheckerSpecResponse)
+		return
+	}
+
+	cm := new(v1.ConfigMap)
+	if len(configMaps.Items) != 1 {
+		logger.Error(err)
+		updateCheckerSpecResponse.Error = "failed to find singular kotsadm-confg config map"
+		JSON(w, 500, updateCheckerSpecResponse)
+		return
+	} else {
+		cm = &configMaps.Items[0]
+	}
+
+	if util.IsHelmManaged() {
+		release := helm.GetHelmApp(mux.Vars(r)["appSlug"])
+		license, err := helm.GetChartLicenseFromSecretOrDownload(release)
+		if err != nil {
+			updateCheckerSpecResponse.Error = "failed to get license from secret"
+			JSON(w, http.StatusInternalServerError, updateCheckerSpecResponse)
+			return
+		}
+		if license == nil {
+			w.WriteHeader(http.StatusNotFound)
+			return
+		}
+
+		if release.GetIsAirgap() {
+			logger.Error(errors.New("airgap scheduled update checks are not supported"))
+			updateCheckerSpecResponse.Error = "airgap scheduled update checks are not supported"
+			JSON(w, 400, updateCheckerSpecResponse)
+			return
+		}
+
+		// validate cron spec
+		cronSpec := configureAutomaticUpdatesRequest.UpdateCheckerSpec
+		if cronSpec != "@never" && cronSpec != "@default" {
+			_, err := cron.ParseStandard(cronSpec)
+			if err != nil {
+				logger.Error(err)
+				updateCheckerSpecResponse.Error = "failed to parse cron spec"
+				JSON(w, 500, updateCheckerSpecResponse)
+				return
+			}
+		}
+
+		cm.Data[fmt.Sprintf("%s-spec", release.GetID())] = cronSpec
+		_, err = clientSet.CoreV1().ConfigMaps(util.PodNamespace).Update(context.TODO(), cm, metav1.UpdateOptions{})
```

Comment:
If config map doesn't exist, this will fail.  This functionality can remain in store package and then all app types can use it.  There are some store functions that use configmaps and secrets already: https://github.com/replicatedhq/kots/blob/6502604a16cac170434a56e19257fbe33c73c97d/pkg/store/kotsstore/session_store.go#L198